src/common/payment.ts

```diff
@@ -233,16 +244,13 @@ export class TurboAuthenticatedPaymentService
     this.tokenTools = tokenTools;
   }
 
-  public async getBalance(): Promise<TurboBalanceResponse> {
-    const headers = await this.signer.generateSignedRequestHeaders();
-    const balance = await this.httpService.get<TurboBalanceResponse>({
-      endpoint: '/balance',
-      headers,
-      allowedStatuses: [200, 404],
-    });
+  public async getBalance(address?: string): Promise<TurboBalanceResponse> {
+    address ??= ownerToNativeAddress(
+      toB64Url(await this.signer.getPublicKey()),
+      this.token,
+    );
```

Comment:
can't this be `this.signer.getNativeAddress` with the changes in https://github.com/ardriveapp/turbo-sdk/pull/142